### PR TITLE
ceph.spec.in: /var/run/ceph fixes

### PR DESCRIPTION
**Pull request by smithfarm**
On systemd distros we should not be creating /var/run/ceph at all because: (1) `/var/run` is symlinked or bind mounted to `/run`, and (2) `/run` is tmpfs (re-created at every boot).

See https://lists.fedoraproject.org/pipermail/devel/2011-March/150031.html

Plus some related specfile cleanups.

http://tracker.ceph.com/issues/12201 Fixes: #12201

Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
hmm, not as simple as I thought

**Comment by smithfarm** [comment]
@ktdreyer @osynge pls review

File: ceph.spec.in [hunk 1]
```diff
@@ -584,7 +585,9 @@ mv $RPM_BUILD_ROOT/sbin/mount.fuse.ceph $RPM_BUILD_ROOT/usr/sbin/mount.fuse.ceph
 
 #set up placeholder directories
 mkdir -p $RPM_BUILD_ROOT%{_sysconfdir}/ceph
+%if ! 0%{?_with_systemd}
```
Comment (osynge) [inline ceph.spec.in]: I would prefer you added a sysVinit variable to the spec file. 

(This is some thing I have been thinking of doing we will see what ken thinks of this idea.)

File: ceph.spec.in [hunk 2]
```diff
@@ -725,7 +728,9 @@ fi
 %dir %{_localstatedir}/lib/ceph/bootstrap-osd
 %dir %{_localstatedir}/lib/ceph/bootstrap-mds
 %dir %{_localstatedir}/lib/ceph/bootstrap-rgw
+%if ! 0%{?_with_systemd}
 %dir %{_localstatedir}/run/ceph/
```
Comment (osynge) [inline ceph.spec.in]: same opinion as above.

**Comment by smithfarm** [comment]
We will also need to call `systemd-tmpfiles --create` in `%post` but that is part of @osynge 's systemd mods

**Comment by smithfarm** [comment]
@dachary The ubuntu make check bot failure says:

```
pip._vendor.requests.exceptions.ConnectionError: HTTPSConnectionPool(host='pypi.python.org', port=443): Max retries exceeded with url: /packages/source/d/distribute/distribute-0.7.3.zip (Caused by <class 'ConnectionResetError'>: [Errno 104] Connection reset by peer)
```

**Comment by dachary** [comment]
@smithfarm I'll take a look, thanks for the ping

**Comment by smithfarm** [comment]
Imagine, I even vaguely remember reading about the double `%%` in comments somewhere. Bad commit removed.

**Comment by smithfarm** [comment]
> On SUSE it's customary to remove the directories from the %files lists and to run 
> systemd-tmpfiles --create /usr/lib/tmpfiles.d/<file_name> during %post.
> 
> On Fedora, it's customary to just create the directories using the %files lists.

So it sounds like I need to change the conditional in this case.

**Comment by smithfarm** [comment]
> I thought the "-D" argument to install does this already?

I guess so - dropping this commit, too.

**Comment by smithfarm** [comment]
Modified `_with_systemd` conditional as per @ktdreyer suggestion.

As for the `%if ! 0%{?_with_systemd}`, would it be reasonable to write it this way:

```
%if (! 0%{?_with_systemd}) || 0%{?fedora}
```

?

**Comment by smithfarm** [comment]
@ktdreyer ping, fixed the conditional

**Comment by ktdreyer** [comment]
Looks good to me - @branto1 care to review?

**Comment by branto1** [comment]
Sure, looks good to me, you can consider this

`Reviewed-by: Boris Ranto <branto@redhat.com>`

**Pull request merged**